### ISSUE TITLE
provider/aws ASG termination_policies changes force new resources

### ISSUE THREAD

**Issue by dlsniper** [hashicorp/terraform#2931]
Hi,

I have the following situation inside an ```aws_autoscaling_group``` when performing a change in the termination policies:

```
termination_policies.#:          "0" => "3" (forces new resource)
termination_policies.3089922268: "" => "ClosestToNextInstanceHour" (forces new resource)
termination_policies.3602983927: "" => "OldestLaunchConfiguration" (forces new resource)
termination_policies.912102603:  "" => "OldestInstance" (forces new resource)
```

This change should not require a new ASG as it can be applied in place.
I will gladly work on this when I'll have time but any pointers on how to get it done would also be great.

Thank you.

**Comment by pforman** [comment]
I had the same issue.  I wrote PR #2890 for it.  The details for why it breaks (basically "update" was never written) are in there.

It also fixes the fact that Terraform can incorrectly reorder your policies if you have more than one.

**Comment by stack72** [comment]
@catsby this can now be closed as #2890 was merged

**Comment by catsby** [comment]
Thanks for the cleanup work, @stack72 :smile:
